### ISSUE TITLE
Training results are good but testing results are very blurred.

### ISSUE THREAD

**Issue by happsky** [junyanz/pytorch-CycleGAN-and-pix2pix#555]
Here are some training results:
![image](https://user-images.githubusercontent.com/5948971/53522499-1c91b400-3aa0-11e9-855d-6c2136a8b4fc.png)

Here are some testing results:
![image](https://user-images.githubusercontent.com/5948971/53522558-3d5a0980-3aa0-11e9-8cc1-b7822dca5d43.png)
![image](https://user-images.githubusercontent.com/5948971/53522577-48ad3500-3aa0-11e9-9d75-7aafa1a11799.png)

My commands are:

```
python train.py --dataroot /data/pytorch-CycleGAN-and-pix2pix_sg2/datasets/ --name ablation_1 --model pix2pix --which_model_netG resnet_9blocks --which_direction AtoB --dataset_mode aligned --norm instance --gpu_ids 0 --batchSize 4 --loadSize 286 --fineSize 256 --no_flip --display_id 1;

python test.py --dataroot /data/pytorch-CycleGAN-and-pix2pix_sg2/datasets/ --name ablation_1 --model pix2pix --which_model_netG resnet_9blocks --which_direction AtoB --dataset_mode aligned --norm instance --gpu_ids 0 --batchSize 4 --loadSize 286 --fineSize 256 --no_flip --eval;

```
Any suggestions, and thanks in advance!

**Comment by happsky** [comment]
Remove `--eval` solved my issue, but I do not know the reason.

**Comment by taesungp** [comment]
It's because dropout and batch norm have different behavior between train and test time, and it is a tricky issue.

**Comment by Ha0Tang** [comment]
But I am using instance norm.

**Comment by John1231983** [comment]
@taesungp : But he removed `--eval` and issue solved. It means that dropout and batch norm have behavior as training. Am I right? I think dropout must be zero and batchnorm must do as the evaluation

**Comment by a1302z** [comment]
I might have a very similar problem, however removing the eval flag does not help there. Is there anything during training or testing that can be done to avoid this?

Just as reference these are my training/test commands
`python train.py --dataroot datasets/AachenDayNight/ --model cycle_gan --name aachendaynight2 --display_port 8834 --display_freq 200 --batch_size 1 --niter 10 --niter_decay 20 --netG unet_128 --no_dropout`
`python test.py --dataroot datasets/AachenDayNight --name aachendaynight2 --model test --no_dropout --netG unet_128 --phase train --num_test 100`
This was one of the last fake images generated during training stored in the web directory
![epoch029_fake_B](https://user-images.githubusercontent.com/15774442/57923504-c5cb9780-78a2-11e9-926f-77203fd7cc1c.png)
And here's the result of the test script (on images from the training set)
![1_fake_B](https://user-images.githubusercontent.com/15774442/57923594-00cdcb00-78a3-11e9-86af-b876e209cfa1.png)

Any advice for this?

**Comment by junyanz** [comment]
Your test script might have some issues.  In your case,  it would be better to use the flag `--model cycle_gan` during test time.
